### PR TITLE
clean up nix expressions

### PR DESCRIPTION
**Pull request by fortuneteller2k**
Please follow this template, if applicable.

## Description

tldr
  - update the lockfile, it's months old already
  - remove the old default* attributes
  - refactor some hacks like that pkgs attribute
  - actually make nix flake check happy

Provide a short description of the changes your PR introduces.
This includes the actual feature you are adding,
as well as any other relevant additions that were necessary to implement your feature.

## Usage

When adding a widget or anything else that affects the configuration,
please provide a minimal example configuration snippet showcasing how to use it and

### Showcase

When adding widgets, please provide screenshots showcasing how your widget looks.
This is not strictly required, but strongly appreciated.

## Additional Notes

Anything else you want to add, such as remaining questions or explanations.

## Checklist

Please make sure you can check all the boxes that apply to this PR.

- [ ] All widgets I've added are correctly documented.
- [ ] I added my changes to CHANGELOG.md, if appropriate.
- [ ] The documentation in the `docs/content/main` directory has been adjusted to reflect my changes.
- [ ] I used `cargo fmt` to automatically format all code before committing

### CODE REVIEW

**Review by fufexan — APPROVED** [review]
LGTM. Can be merged.
